WebTransportSendStream docs plus WebTransportReceiveStream and friends (#30486)
mdn/content: files/en-us/web/api/webtransport/createunidirectionalstream/index.md

The **`createUnidirectionalStream()`** method of the {{domxref("WebTransport")}} interface asynchronously opens a unidirectional stream.

> The method returns a {{jsxref("Promise")}} that resolves to a {{domxref("WebTransportSendStream")}} object.
> This object inherits from {{domxref("WritableStream")}}, and can be used to reliably write data to the server, and to query stream statistics and other information.

Author: ```suggestion
The method returns a {{jsxref("Promise")}} that resolves to a {{domxref("WritableStream")}} object, which can be used to reliably write data to the server.
<!-- Note, returns a `WebTransportSendStream` according to spec, but not yet implemented -->
```

### Return value

> A {{jsxref("Promise")}} that resolves to a {{domxref("WebTransportSendStream")}} object.

Author: NOTE, on FF this returns a `WritableStream`! However the spec says it should return a `WebTransportSendStream`. Further FF does use the SendStream for the bidirectional case.

IT is LIKELY that Chrome does this properly but I can't test this (none of the tests I try on WPT seem to do handshake).

Author: Reverting
```suggestion
A {{jsxref("Promise")}} that resolves to a `WebTransportSendStream` object (this is a {{domxref("WritableStream")}}).
```

Note that this object inherits from {{domxref("WritableStream")}}.

> Implementations based on older versions of the specficiatoion may return a `WritableStream` instead.

Author: ```suggestion
Implementations based on older versions of the specification may return a `WritableStream` instead.
```

> 
> Note that this object inherits from {{domxref("WritableStream")}}.
> Implementations based on older versions of the specification may return a `WritableStream` instead.

Author: Reverting
```suggestion
```

## Examples

> Use the `createUnidirectionalStream()` method to get a reference to a {{domxref("WebTransportSendStream")}}.
> From this you can {{domxref("WritableStream.getWriter", "get a writer", "", "nocode")}} to allow data to be written to the stream and sent to the server.

Author: Reverting
```suggestion
Use the `createUnidirectionalStream()` method to get a reference to a {{domxref("WritableStream")}}. From this you can {{domxref("WritableStream.getWriter", "get a writer", "", "nocode")}} to allow data to be written to the stream and sent to the server.
```
